src/appReview/saga.ts

```diff
@@ -0,0 +1,31 @@
+import InAppReview from 'react-native-in-app-review'
+import { call, put, select, spawn, takeLatest } from 'redux-saga/effects'
+import { lastInteractionTimestampSelector } from 'src/appReview/selectors'
+import { setLastInteractionTimestamp } from 'src/appReview/slice'
+import { Actions as SendActions } from 'src/send/actions'
+import { safely } from 'src/utils/safely'
+import { ONE_DAY_IN_MILLIS } from 'src/utils/time'
+
+export function* setAppReview() {
+  // Quick return if the device does not support in app review
+  if (!InAppReview.isAvailable()) return
+
+  const lastInteractionTimestamp = yield select(lastInteractionTimestampSelector)
+  const now = Date.now()
+
+  // If the last interaction was less than a quarter year ago or null
+  if (lastInteractionTimestamp <= now - ONE_DAY_IN_MILLIS * 91) {
```

Comment:
nit: it'd be nice to have the threshold as a constant at the top of the file so that it's easy to find / update. so then you could have `if (now - lastInteractionTimestamp >= REVIEW_INTERVAL)`?

also, i think if apple only allows 3 review per year then the interval should be 122 days?

Reply:
App Store policy governs the actual display of a rating and review request view from [`requestReview(in:)`](https://developer.apple.com/documentation/storekit/skstorereviewcontroller/3566727-requestreview#discussion) [used by react-native-in-app-review](https://github.com/MinaSamir11/react-native-in-app-review#system-rating-and-review-prompts). Android seems to be [limited to about once per month](https://developer.android.com/guide/playcore/in-app-review#:~:text=Play%20Store%20mechanism.-,Quotas,not%20always%20display%20a%20dialog.), but can change anytime. As such once a quarter feels pretty unintrusive considering behavior I've seen in other apps.

Reply:
Adjusted the style of the check in abdc93177a7e5e509055a5f6d7d173109a639379!